Runtime/Utility/CountdownTimer.cs

```diff
@@ -53,13 +73,59 @@ public bool IsRunning
             protected set;
         }
 
+        /// <summary>
+        /// Elapsed time of the timer.
+        /// </summary>
+        public float ElapsedTime
+        {
+            get
+            {
+                if (IsRunning)
+                {
+                    currentTime = Time.time;
+                }
+                return currentTime - beginTime;
+            }
+        }
+
+        /// <summary>
+        /// Remaining time of the timer.
+        /// </summary>
+        public float RemainingTime
+        {
+            get
+            {
+                if (IsRunning)
+                {
+                    currentTime = Time.time;
+                }
+                return StartTime + (beginTime - currentTime);
+            }
+        }
+
+        /// <summary>
+        /// <see cref="Time.time"/> when <see cref="Begin"/> is called.
+        /// </summary>
+        protected float beginTime;
+        /// <summary>
+        /// <see cref="Time.time"/> of the current frame.
+        /// </summary>
+        protected float currentTime;
+
+        protected virtual void SetInternalStates()

```

Comment:
requires a comment as to what the method does, also protected methods appear after public methods.

So move this before the `Complete` on line 194 then it should be good to go.